src/haxelib/client/Main.hx

```diff
@@ -1037,19 +1049,32 @@ class Main {
 		}
 		else
 		{
-			var inf = try site.infos(p) catch( e : Dynamic ) { Sys.println(e); return; };
-			p = inf.name;
-			if( !FileSystem.exists(pdir+"/"+Data.safe(inf.getLatest())) ) {
+			var latest = try site.getLatestVersion(p) catch( e : Dynamic ) { Sys.println(e); return; };
+
+			if( !FileSystem.exists(pdir+"/"+Data.safe(latest)) ) {
 				if( state.prompt ) {
-					if (!ask("Update "+p+" to "+inf.getLatest()))
+					if (!ask("Update "+p+" to "+latest))
 						return;
 				}
-				doInstall(p,inf.getLatest(),true);
+				doInstall(p, latest, true);
 				state.updated = true;
 			} else
-				setCurrent(p, inf.getLatest(), true);
+				setCurrent(p, latest, true);
 		}
 	}
+
+	function projectNameToDir( project:String, state ) {
+		var p = project.toLowerCase();
+
+		var l = FileSystem.readDirectory(state.rep).filter(function (dir) return dir.toLowerCase() == p);
+
+		switch (l.length) {
```

Comment:
it's not important or anything, but note that this could be written as
```haxe
switch (l) {
    case []: throw ...;
    case [dir]: return Data.unsafe(dir);
    case _: throw ...;
}
```
the generated code will be pretty much the same, but I personally find it more readable :)